### ISSUE TITLE
Current status of the project

### ISSUE THREAD

**Issue by ZaMaZaN4iK** [jinja2cpp/Jinja2Cpp#195]
Hello.

Unfortunately @flexferrum cannot continue future development of Jinja2Cpp (link: https://www.facebook.com/sergey.sadovnikov.35). For english-speaking people: @flexferrum is dead :(

He left awesome projects. One of them is Jinja2Cpp. So the main question for now is: what is the future of Jinja2Cpp? 

The project is a quite popular (according to GitHub stars), so possibly there are people who are interested in maintaining and future development. @AndreyBronin @palchukovsky @Manu343726 - I don't know another people by their nicknames. Possibly you can add something.

If any person is interested in it - I suggest to cooperate in this org and continue at least maintaining.

**Comment by Manu343726** [comment]
I would like to help. Do you know if he left some kind of roadmap written
somewhere?

Also I think we should also discuss about the future of autoprogrammer, his
metaclasses generation project which I think was the reason he created
jinja2cpp in the first place.

**Comment by ZaMaZaN4iK** [comment]
> I would like to help. Do you know if he left some kind of roadmap written
> somewhere?
I don't know about any roadmap. I think good start point is what is written in issues (as far as I see he created a lot of issues with possible improvements).

And I agree about `autoprogrammer` - that's also important project.

I think we can wait a little bit for any updates from other members of `jinja2cpp` org, since, possibly, they also have some ideas.

Will be fine, if development will be continued in this repo. Otherwise migration process should be initialized (mark your fork as main, left here a noticement in issues about that, etc.)

**Comment by Manu343726** [comment]
Iirc he was working on a proof of concept of using the llvm JIT to speedup template processing

**Comment by ZaMaZaN4iK** [comment]
I've heard nothing about this work :( So if it was local unpublished work, it's 99% lost work, I think :(

**Comment by leodido** [comment]
I’d like to help too. In any way I could help.

**Comment by Manu343726** [comment]
Oh, first things first, do we have admin access to the org? To invite people, maybe moving the rest of his projects there, etc?

**Comment by ZaMaZaN4iK** [comment]
No, we haven't. That's why I've pinged @AndreyBronin and @palchukovsky . If no - we need to move interesting for us repos to another org and then left here a remark about new place for Jinja2Cpp and co.

**Comment by martinmoene** [comment]
@ZaMaZaN4iK I'm sorry to hear this sad news of @flexferrum 's passing. Would you like to send me the information from the Facebook post, as I'm not and do not want to be on Facebook? Thanks in advance, Martin (author of the *-lite libraries).

**Comment by blockspacer** [comment]
If anyone interested in development of tools similar to https://github.com/flexferrum/autoprogrammer or https://github.com/blockspacer/CXXCTP - please contact me. I'll be happy to contribute. E-mail: derofim hosted on yandex.ru

**Comment by rmorozov** [comment]
I think it’s good idea to continue Sergey’s work on this project. We can try to analyze current project’s state and move it further.

**Comment by blockspacer** [comment]
I deprecated https://github.com/blockspacer/CXXCTP to create better alternative, currently it is named flextool. flextool is based on https://github.com/flexferrum/autoprogrammer 

Github repo: https://github.com/blockspacer/flextool
Main project page: https://blockspacer.github.io/flex_docs/

**Some interesting features of flextool for contributors**:
- It uses `chromium/base` packaged via conan, so it's codebase may be very interesting. For example, you can add tracing to any C++ project similar to https://aras-p.info/blog/2017/01/23/Chrome-Tracing-as-Profiler-Frontend/
- It uses conan to manage plugins, find executable, LLVM_ROOT, etc.
- Supports Cling C++ interpreter and provides useful lib to communicate from C++ to Cling
- Supports A* promises with sequence checking https://github.com/blockspacer/basis/tree/master/basis/promise

Now i'm improving code quality of flextool and making design decisions. Please open Github issue https://github.com/blockspacer/flextool/issues if you know how to improve flextool.

flextool can be migrated into github organization, so not only 1 person will be able to maintain project. Please contact me if you are interested in contributing. E-mail: derofim hosted on yandex.ru

**Comment by DragonOsman** [comment]
I'm sad to hear about @flexferrum's passing. :(

@ZaMaZan4iK Do you guys have a new repo for the project yet?  Or are you continuing in this repo?  

I have a problem with building the library that I opened an issue on, too.

**Comment by ZaMaZaN4iK** [comment]
@DragonOsman unfortunately we cannot continue work in this repo since we have no permissions for doing such stuff.

Also we didn't create a new repo/fork. Feel free to create your own fork and fix your problems, since this project is unmaintained for now (and seems like no one is interested in maintaining it)

**Comment by DragonOsman** [comment]
@ZaMaZaN4iK I opened an issue about my problem [here](https://github.com/jinja2cpp/Jinja2Cpp/issues/204).  If possible, please look at the error messages I got and guide me in fixing my problem.  Thanks.

**Comment by AntonioCS** [comment]
@ZaMaZaN4iK have you tried reaching out to Github and explaining the situation? I mean, if someone really wants to take ownership of this repo I'm sure github would be understanding due to what happened to the author (at least I hope they are).

**Comment by luismartingil** [comment]
Does anyone know if this process is continued finally? Not sure what would be Github's position here, but it would be nice it could continue being an active project.

**Comment by rmorozov** [comment]
Hello guys!

@flexferrum was a friend of mine and I have rights to accept merge requests. 

I don’t have enough spare time to follow Sergey’s roadmap on this project, but I see no problems to review and accept MRs.

So if someone wants to contribute - don’t hesitate to contact me. I will do my best to perform some maintenance work over this project.

**Comment by luismartingil** [comment]
> Hello guys!
> 
> @flexferrum was a friend of mine and I have rights to accept merge requests.
> 
> I don’t have enough spare time to follow Sergey’s roadmap on this project, but I see no problems to review and accept MRs.
> 
> So if someone wants to contribute - don’t hesitate to contact me. I will do my best to perform some maintenance work over this project.

@rmorozov Please see [this PR](https://github.com/jinja2cpp/Jinja2Cpp/pull/209) when you have a chance

**Comment by blockspacer** [comment]
Created https://github.com/FlexPlusOrg

FlexPlusOrg contains flextool that is based on https://github.com/flexferrum/autoprogrammer

Please contact me if you are interested in contributing. E-mail: derofim hosted on yandex.ru
